src/Temporalio/Worker/TemporalWorker.cs

```diff
@@ -92,7 +92,12 @@ public TemporalWorker(IWorkerClient client, TemporalWorkerOptions options)
                     OnTaskCompleted: options.OnTaskCompleted,
                     RuntimeMetricMeter: MetricMeter,
                     WorkerLevelFailureExceptionTypes: options.WorkflowFailureExceptionTypes,
-                    DisableEagerActivityExecution: options.DisableEagerActivityExecution));
+                    DisableEagerActivityExecution: options.DisableEagerActivityExecution,
+                    AssertValidLocalActivity: (activityType) => (activityWorker ??
+                                throw new InvalidOperationException(
+                                    $"Activity {activityType} is not registered on this worker," +
+                                    $" no available activities."))
+                                .AssertValidActivity(activityType)));
```

Comment:
A bit convoluted for a single-expression lambda, but meh. Feel free to have a private method in this class do this and just pass that. But not required.